pkg/apiserver/handlers/featuregates/handler.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package featuregates
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/antctl/runtime"
+	"github.com/vmware-tanzu/antrea/pkg/features"
+)
+
+var controllerOnlyGates = []string{
+	"Traceflow",
+	"AntreaPolicy",
+	"Egress",
+	"NetworkPolicyStats",
+}
+
+type Response struct {
+	FeatureName    string `json:"name,omitempty"`
+	FeatureStatus  string `json:"status,omitempty"`
+	FeatureVersion string `json:"version,omitempty"`
+}
+
+// HandleFunc returns the function which can handle queries issued by featuregates commands.
+// The handler function populates Antrea featuregates information to the response.
+func HandleFunc() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		isController := false
+		if runtime.Mode == runtime.ModeController {
+			isController = true
+		}
```

Comment:
This is more of an antctl concept. I'd rather have 2 different handlers or add a parameter to this handler function.